modules/integration/tests-integration/lazy-loading-test/src/test/java/org/wso2/appserver/integration/lazy/loading/LazyLoadingBaseTest.java

```diff
@@ -0,0 +1,516 @@
+     * status information.
+     * IT will return the  the deployment status of given web application of given tenant is loaded in Ghost form.
+     *
+     * @param tenantDomain -  Domain Name of the tenant.
+     * @param webAppName   -  Name of the Web Application
+     * @return WebAppStatus -  which include Tenant status, whether the web-app is started or not, whether the web-app is in
+     * ghost mode or not.
+     * @throws LazyLoadingTestException - Exception throws when send the Get request and retrieve the JSON data from response.
+     */
+    protected WebAppStatus getWebAppStatus(String tenantDomain, String webAppName) throws LazyLoadingTestException {
+        String requestUrl = webAppURL + "/" + IS_WEB_APP_LOADED_METHOD_URL + "/" + tenantDomain + "/" + webAppName;
+        WebAppStatus webAppStatus;
+        try {
+
+            HttpResponse response = HttpURLConnectionClient.sendGetRequest(requestUrl, null);
+            JSONObject webAppStatusJSON = new JSONObject(response.getData()).getJSONObject("WebAppStatus");
+            boolean isTenantLoaded = webAppStatusJSON.getJSONObject("tenantStatus").getBoolean("tenantContextLoaded");
+            boolean isWebAppStarted = webAppStatusJSON.getBoolean("webAppStarted");
+            boolean isWebAppGhost = webAppStatusJSON.getBoolean("webAppGhost");
+            webAppStatus = new WebAppStatus(new TenantStatus(isTenantLoaded), isWebAppStarted, isWebAppGhost);
+        } catch (IOException ioException) {
+            String customErrorMessage =
+                    "IOException when sending the Get request to:" + requestUrl;
+            log.error(customErrorMessage, ioException);
+            throw new LazyLoadingTestException(customErrorMessage, ioException);
+        } catch (JSONException jsonException) {
+            String customErrorMessage = "JSONException when retrieving the values from json object WebAppStatus" + requestUrl;
+            log.error(customErrorMessage, jsonException);
+            throw new LazyLoadingTestException(customErrorMessage, jsonException);
+        }
+
+        return webAppStatus;
+
+    }
+
+
+    /**
+     * Check the given Jaggery application is deployed correctly. This method is wait for 90 seconds
+     * for deployment of jaggery application and each 500 milliseconds  of wait it will check the
+     * deployment status.
+     *
+     * @param appName - Name of the application.
+     * @return boolean - true if the application is get deployed before the maximum wait time of 90 seconds.
+     * @throws LazyLoadingTestException - Exception throws when creating WebAppAdminClient.
+     */
+    protected boolean isJaggeryAppDeployed(String appName) throws LazyLoadingTestException {
+        int deploymentDelayInMilliseconds = 90 * 1000;
+        log.info("waiting " + deploymentDelayInMilliseconds + " millis for Service deployment " + appName);
```

Comment:
Better move this log line to the end of variable declaration. It is bit hard to read variables when it mixed with log messages.